src/game/enemy/mod.rs

```diff
@@ -1,27 +1,84 @@
 use bevy::prelude::*;
 
 mod components;
-mod systems;
 pub mod events;
+mod resources;
+mod systems;
 
 use systems::*;
 
+use crate::components::AnimationTimer;
 use crate::AppState;
-use crate::game::enemy::events::EnemyShotEvent;
+use crate::{components::AnimationIndices, game::enemy::events::EnemyShotEvent};
+
+use self::events::EnemyShotPlayerEvent;
+use self::resources::{EnemySpawnTimer, EnemySpawns};
+use super::InGameState;
 
 pub struct EnemyPlugin;
 
 impl Plugin for EnemyPlugin {
     fn build(&self, app: &mut App) {
         app
+            // Events
             .add_event::<EnemyShotEvent>()
-            // Start up systems
+            .add_event::<EnemyShotPlayerEvent>()
+            // Startup system
+            .add_systems(Startup, init_texture_atlas_handles)
+            // Systems
+            .add_systems(Update, init_enemy_level_info)
```

Comment:
shouldnt this only be needed when entering a level?

Reply:
It uses an eventreader, so I assumed it was already doing that. I'll add it to be explicit